### ISSUE TITLE
Delete user leaves DB residue - page has uncaught reference error 

### ISSUE THREAD

**Issue by EzraCc** [opensource-socialnetwork/opensource-socialnetwork#1037]
OSSN 4.3.1
PHP 7.0.15
Apache 2.4.18
Chrome console error: Uncaught ReferenceError: OSSN is not defined.
PHP Stack Trace, error starts at profileLink();
(Apologies, I cleaned my DB tables and fixed the errors without taking screen shots. The site is not publicly available.)

Steps to reproduce:
1. Create user
2. Do activity as user: posts, friend, comments on others, likes, etc..
3. Sign out, sign in as admin
4. Check ossn_users.guid for the user in DB, if possible
5. Delete user via page: /administrator/users
6. View Site
7. Proceed through feed pagination, if required, to reach point where posts the deleted user liked/commented appear.
8. Pages break. The functions to load user info and links of deleted user fail. Error is not handled by PHP code.

I looked through the DB tables and found entries in these two tables where the owner_guid matched the deleted users:
- ossn_entities
- ossn_object

ossn_entities.active for those users remained =1.
ossn_object does not have an 'active' column.

Going in and deleting those entries from both tables resolved the problem. 

There could have been entries in other tables (messages, notifications, relationships) for the deleted user as well. In my case there weren't.

The delete user function likely needs to go through any table that uses ossn_user.guid as a foreign key and either delete them or mark them as inactive. In some cases (ossn_object), the latter would require adding the active column.)

FWIW - I have seen both states in other Blogging and Social Media sites, where they initially flag a user and their content as inactive first but leave it in a retrievable state. After additional user/admin input or time, they delete the content altogether. That would require making sure all queries had an AND active=1 clause.

**Comment by lianglee** [comment]
Can you tell if you are using any 3rd party component?

**Comment by EzraCc** [comment]
I did (sorry), none of them should have affected this, but only one way to find out. I'll do a fresh install that I can break, and recreate to verify.

Components
Auto Validate Users 3.1 (users deleted were validated via this method)
Fake Users 2.0 (users deleted were not created with this)
Fake Postings 1.0 (users deleted had no posts created by this)
Russian language pack (not a component, but is an add-on)

**Comment by githubertus** [comment]
Do you remember of wich `type/subtype` these orphan records were?

**Comment by EzraCc** [comment]
I'm going through a fresh install environment with no components to thoroughly investigating the tables for orphans and errors after each type of delete: post / comment / user. It'll be a few days before I'm able to get back to it. I'll post full table:row details w/ screenshots when I'm able to do that.

I believe there are orphan rows in entities_metadata after a post delete, I will verify (the entities rows were properly deleted). I will verify per above.

I noticed that for posts set to friends only (access=3), if another user sends a friend request those posts become visible before the other user accepts the request, which shouldn't be the case. The check there needs to verify the relationships is reciprocal before displaying friends only content.

**Comment by githubertus** [comment]
Thank you very much for your helpful investigations.
As for the privacy issue (good catch!): I'll open a new thread for that one in order to keep (and solve) separate problems separately.

**Comment by githubertus** [comment]
@EzraCc 
Can you give any new input regarding this issue?
Actually, I have tried all kind of actions yesterday in order to reproduce the problem: without success. Maybe it needs a very special sequence of actions to trigger the problem ... I have no idea ...

One question: Are you really running a plain vanilla Ossn 4.3.1 (https://www.opensource-socialnetwork.org/downloads/ossn-v4.3.1-1480759958.zip) or perhaps a development version from here?

**Comment by EzraCc** [comment]
I've been delayed by other work; I'll be able to complete tests on this in the next couple of days. I have done a fresh install of plain vanilla OSSN 4.3.1 to use for those tests, from ossn.org not github.

**Comment by EzraCc** [comment]
Apologies for the delay (and bad bug report). I completed my tests on a fresh install of OSSN 4.3.1 with no components added, checking the UI and DB records via SSH after user deletion and was not able to replicate. 

I verified that records were removed from the following tables.  
- ossn_object
- ossn_likes
- ossn_relationships
- ossn_annotations
- ossn_entities
- ossn_entities_metadata
- ossn_messages

I also verified photos uploaded by the users were deleted.

A separate issue was identified and replicated from my tests. I'll post a new thread to keep them separate.
